src/core/component.js

```diff
@@ -487,6 +489,24 @@ module.exports.registerComponent = function (name, definition) {
 };
 
 /**
+* Clones component data.
+* Clones only the properties that are plain objects while
+* keeping a reference for the rest.
+*
+* @param data - Component data to clone.
+* @returns Cloned data.
+*/
+function cloneData (data) {
+  var clone = {};
+  var parsedProperty;
+  for (var key in data) {
```

Comment:
declare vars up top.

i wonder if we should start to use `for (key in obj)` for every loop over an object where we know it's a plain JS object. e.g., change some of the code in the `buildData` to also use `for...in`